### PR TITLE
fix(engine-odim): dedup PVOL VerticalProfile z-axis elevation angles

### LINKED ISSUE #269: odim-volume EDR: VerticalProfile z-axis has duplicate elevation angles → invalid CoverageJSON
## Symptom

An EDR `position` (or `locations`/`area`) query against a PVOL collection with no `z` returns CoverageJSON that fails schema validation.

Example:
```
GET /edr/collections/radar-fi-volume-local-h5/position?f=CoverageJSON&parameter-name=DBZV&datetime=2026-05-15T00:00:00Z&coords=POINT(26.031 61.306)
```

The response is HTTP 200 but the `VerticalProfile` `z` axis contains a duplicate value:
```
z: [0.3, 0.7, 1.5, 2.0, 2.0, 3.0, 5.0, 7.0, 9.0, 11.0, 15.0, 25.0, 45.0]
```

CoverageJSON axis values are `uniqueItems` (`valuesAxisBase` in `schemas/coveragejson.json`), so the document is invalid.

## Root cause

FMI radars run **split cuts** — two sweeps at the same nominal elevation angle (e.g. a surveillance cut carrying `DBZH` and a Doppler cut carrying `VRADH`). `volume_profile()` in `crates/engine-odim/src/volume_engine.rs` built the `z` axis straight from every sweep with no dedup, while the catalog's collection-level vertical extent already deduped — so the per-coverage axis and the advertised extent were inconsistent.

## Fix

Collapse the per-sweep angles to distinct values (sort + dedup, matching the catalog), and sample each quantity per unique angle from whichever sweep at that angle carries it. Split cuts then resolve correctly: `DBZH` from the surveillance cut, `VRADH` from the Doppler cut.

## Verification

- Regression test asserting a deduped `z` axis and matching range shape.
- Live re-query: 12 unique `z` values, range shape `[12]`, all 16 PVOL parameters pass dup-axis + shape-product checks.

### PR DESCRIPTION
**Pull request by mrauhala**
## Summary

- Fixes #269 — a PVOL EDR `position`/`locations`/`area` query with no `z` returned CoverageJSON whose `VerticalProfile` `z` axis had a duplicate elevation angle, violating the schema's `uniqueItems` constraint on axis values.
- Root cause: FMI radars run split cuts (two sweeps at the same nominal angle, e.g. a surveillance cut with `DBZH` and a Doppler cut with `VRADH`). `volume_profile()` built the `z` axis straight from every sweep with no dedup, while the catalog's collection-level vertical extent already deduped.
- Fix: collapse per-sweep angles to distinct values (sort + dedup, matching the catalog) and sample each quantity per unique angle from whichever sweep at that angle carries it — so split cuts resolve correctly.

## Test plan

- [x] New regression test `volume_profile_dedups_split_cut_elevation_angles` (4 sweeps, two at 2.0° carrying different moments) asserts a deduped `z` axis and matching range shape.
- [x] `cargo test -p engine-odim` green (74 + 19 tests); `cargo fmt` + `cargo clippy -D warnings` clean.
- [x] Live re-query of the reported URL on a fresh build: 12 unique `z` values, range shape `[12]`, all 16 PVOL parameters pass dup-axis + shape-product checks.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by mrauhala** [comment]
Addressed the review findings in 0efad8a:

1. **`find_map` nodata vs quantity-absent** — intentional, kept. For a single sweep the result is identical to before (`None` when the quantity is absent or nodata). For a split cut (two sweeps at the same rounded angle), falling through to the sibling yields the available measurement instead of a gap, which is the desired behavior for a single-point profile. Documented the fall-through in a comment, and the strengthened test (below) pins it down.

2. **`Vec::dedup()` + NaN** — fixed. `dedup()` uses `PartialEq`, so a `NaN` elevation (which `total_cmp` sorts together) would not collapse. Switched both the `volume_profile` z-axis and the catalog vertical-extent dedup to `dedup_by(|a, b| a.total_cmp(b).is_eq())`. (A deeper "reject non-finite elangle at read time" is a possible follow-up; elevation angles are finite in real ODIM data.)

3. **Test asserted only shape/length** — fixed. The test now queries `DBZH` + `VRADH` together and asserts the value pattern: `DBZH` is sampled at every level, while `VRADH` (carried only by the Doppler cut at 2.0°) is populated *only* at z=2.0 with `None` at 0.5°/5.0° — proving the per-angle search selects per-quantity rather than "first sweep at the angle wins".

File: crates/engine-odim/src/volume_engine.rs [hunk 1]
```diff
@@ -1237,24 +1239,45 @@ fn snap_levels(requested: &[f64], canonical: &[f64]) -> Vec<f64> {
...
             .iter()
-            .map(|sweep| {
-                let moment = sweep.moments.iter().find(|m| &m.quantity == quantity)?;
-                sample_sweep_moment(sweep, moment, site_lon, site_lat, lon, lat)
+            .map(|&level| {
+                entry
+                    .volume
+                    .sweeps
+                    .iter()
+                    .filter(|s| round_elevation(s.elangle) == level)
```
[the remark this answers is not in the document]
Comment (mrauhala) [reply crates/engine-odim/src/volume_engine.rs]: Fixed in 3c1ee12 — applied the suggestion, the filter now uses `round_elevation(s.elangle).total_cmp(&level).is_eq()` so it stays consistent with the dedup. Agreed it's theoretical (NaN angle ⇒ malformed file), but the two predicates should match.

**Comment by mrauhala** [comment]
Round 3 review:

- **NaN elangle reaches the z-axis (CONFIRMED)** — fixed in f1a91c4. Even though dedup collapsed duplicate NaNs, a NaN still flowed into the `VerticalProfile` z-axis (and the catalog vertical extent) where it serialises to JSON `null` and breaks the numeric axis. Now both sites `retain(|v| v.is_finite())` before sort/dedup, so non-finite angles are dropped entirely. Added a `volume_profile_drops_nan_elevation` regression test.

- **`find_map` conflates "quantity absent" with "nodata" (PLAUSIBLE, re-cited)** — intentional, unchanged (see the earlier resolution + the `find_map` comment at the call site). For a single sweep the result is identical to before (`None` when absent or nodata); for a split cut, falling through to the carrying sibling returns the available measurement instead of a gap. The `volume_profile_dedups_split_cut_elevation_angles` test pins this down (VRADH populated only at the cut that carries it).

**Comment by mrauhala** [comment]
Round 5: empty z axis when every elangle is non-finite — fixed in 4381f28. `volume_profile` now returns `Option<QueryResult>` (None when no finite sweep remains after the `retain(is_finite)`), and `site_coverages` uses `filter_map` so a severely malformed all-NaN volume is dropped rather than emitting a `VerticalProfile` with an empty z axis (which the CoverageJSON schema would reject — `numericValuesAxis.values` has `minItems: 1`). Added an all-NaN regression test.

**Closing the review loop on this PR.** The original duplicate-z bug is fixed and locked down with regression tests; the subsequent rounds tightened NaN/`-0.0`/split-cut/empty-volume edge cases (each round addressed at the root in the next commit, not band-aided). CI is green, substance is solid. Further rounds re-citing the same already-resolved-or-by-design items would be the stateless reviewer's diminishing returns rather than new issues.
